weave-js/src/components/PagePanelComponents/Home/Browse3/filters/types.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  GridColDef,
+  GridColumnGroupingModel,
+  GridFilterItem,
+  GridFilterModel,
+} from '@mui/x-data-grid-pro';
+import _ from 'lodash';
+
+import {TraceCallSchema} from '../pages/wfReactInterface/traceServerClientTypes';
+
+export type FilterId = number | string | undefined;
+
+// These are columns we won't allow the user to filter on.
+// For most of these it would be great if we could enable filtering in the future.
+export const UNFILTERABLE_FIELDS = [
+  'op_name',
+  'feedback',
+  'summary.weave.status',
+  'summary.weave.tokens',
+  'summary.weave.cost',
+  'summary.weave.latency',
+  'wb_user_id', // Option+Click works
+];
+
+export type ColumnInfo = {
+  cols: Array<GridColDef<TraceCallSchema>>;
+  colGroupingModel: GridColumnGroupingModel;
+};
+
+export const FIELD_LABELS: Record<string, string> = {
+  id: 'Call ID',
+  started_at: 'Called',
+  wb_user_id: 'User',
+};
+
+export const getFieldLabel = (field: string): string => {
+  return FIELD_LABELS[field] ?? field;
+};
+
+export const FIELD_TYPE: Record<string, string> = {
+  id: 'id',
+  'summary.weave.status': 'status',
+  wb_user_id: 'user',
+  started_at: 'datetime',
+};
+
+export const getFieldType = (field: string): string => {
```

Comment:
this file seems to have a lot of overlap with the `operators.ts` file - any use in sharing some code?

Reply:
You're right, I actually think we can get rid of a lot of operators.ts with this PR.